dotCMS/src/main/java/com/dotcms/rest/api/v1/system/ConfigurationResource.java

```diff
@@ -48,13 +59,92 @@
 
 	private final ConfigurationHelper helper;
 
+	private static final Pattern BLACK_LIST = Try.of(()->Pattern.compile(
+			Config.getStringProperty("OBFUSCATE_SYSTEM_ENVIRONMENTAL_VARIABLES", "passw|pass|passwd|secret|key|token"),
+			Pattern.CASE_INSENSITIVE)).getOrNull();
+
+	private static final Set<String> WHITE_LIST = ImmutableSet.copyOf(
+			Config.getStringArrayProperty("CONFIGURATION_WHITE_LIST",
+					new String[] {"EMAIL_SYSTEM_ADDRESS", "CHARSET","CONTENT_PALETTE_HIDDEN_CONTENT_TYPES" }));
+
+
+
+	private boolean isOnBlackList(final String key) {
+
+		return null != BLACK_LIST? BLACK_LIST.matcher(key).find() : false;
+	}
 	/**
 	 * Default constructor.
 	 */
 	public ConfigurationResource() {
 		this.helper = ConfigurationHelper.INSTANCE;
 	}
 
+	/**
+	 * Retrieve the keys from dotcms Configuration (allowed on WHITE_LIST and are not restricted by the BLACK_LIST)
+	 * @param request
+	 * @param response
+	 * @param keysQuery
+	 * @return
+	 * @throws IOException
+	 */
+	@Path("/config")
+	@GET
+	@JSONP
+	@NoCache
+	@Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+	public final Response getConfigVariables(@Context final HttpServletRequest request,
+											 @Context final HttpServletResponse response,
+											 @QueryParam("keys") final String keysQuery)
+			throws IOException {
+
+		new WebResource.InitBuilder(request, response)
+				.requiredBackendUser(true)
+				.requestAndResponse(request, response)
+				.rejectWhenNoUser(true)
+				.init();
+
+		final String [] keys = StringUtils.splitByCommas(keysQuery);
+		final Map<String,Object> resultMap = new LinkedHashMap<>();
+		if (null != keys) {
+
+			for (final String key : keys) {
+
+				final String keyWithoutPrefix = this.removePrefix (key);
+				if (this.WHITE_LIST.contains(keyWithoutPrefix) && !this.isOnBlackList(keyWithoutPrefix)) {
+
+					resultMap.put(keyWithoutPrefix, recoveryFromConfig(key));
+				}
+			}
+		}
+
+		return Response.ok(new ResponseEntityView(resultMap)).build();
+	}
+
+	private String removePrefix (final String key) {
+
+		return key.replace("list:", StringPool.BLANK)
+				.replace("boolean:", StringPool.BLANK)
+				.replace("number:", StringPool.BLANK);
+	}
+
+
+	private Object recoveryFromConfig (final String key) {
+
+		if (key.startsWith("list:")) {
+
+			return Arrays.asList(Config.getStringArrayProperty(key.replace("list:", StringPool.BLANK), new String[]{}));
+		} else if(key.startsWith("boolean:")) {
+
+			return Config.getBooleanProperty(key.replace("boolean:", StringPool.BLANK), false);
+		} else if (key.startsWith("number:")) {
+
+			return Config.getIntProperty(key.replace("boolean:", StringPool.BLANK), 0);
```

Comment:
The replace here shouldn't be: `Config.getIntProperty(key.replace("number:", StringPool.BLANK), 0);` ?